### PR TITLE
Adds  ability to override cols with custom col based on field

### PR DESCRIPTION
**Pull request by ciaranschutte**

### CODE REVIEW

**Comment by ciaranschutte** [comment]
https://github.com/kids-first/kf-portal-ui/pull/1061

File: modules/components/src/DataTable/utils.js [hunk 1]
```diff
@@ -20,18 +20,33 @@ export function normalizeColumns({
...
-        : column.hasCustomType,
-      ...(!column.accessor && !column.id ? { id: column.field } : {}),
-      ...(customTypeConfigs[column.type] || {}),
-    }))
+    .map(column => {
+      const customColIndex = customColumns.findIndex(
+        cc => cc.content.field === column.field,
+      );
+
+      let customCol = {};
```
Comment (hlminh2000) [inline modules/components/src/DataTable/utils.js]: I think we can achieve this with:
```
const customCol = customColumns.find(cc => cc.content.field === column.field)?.content || {}
```
Then, I don't think there's a need for the `if` bellow, but please correct me if I'm wrong :P

File: modules/components/src/DataTable/utils.js [hunk 2]
```diff
@@ -20,18 +20,33 @@ export function normalizeColumns({
...
+      if (customColIndex > -1) {
+        customCol = customColumns[customColIndex].content;
+        customColumns.splice(customColIndex, 1);
```
Comment (hlminh2000) [inline modules/components/src/DataTable/utils.js]: This mutation feels troubling to me, especially in this case as `customColumns` is then used outside of this map. It's better imo to compute a new filtered array outside the map for use later.

Comment (ciaranschutte) [reply modules/components/src/DataTable/utils.js]: So before this mapping create two arrays "custom columns to use overriding" and a "custom columns to add"?

Comment (hlminh2000) [reply modules/components/src/DataTable/utils.js]: I think after makes more sense, a const that's the filtered `customColumns` to feed in here https://github.com/overture-stack/arranger/pull/478/files/aaf642fcd7a51a475ff507fd9a9484c96126e6de#diff-f0ae8de5b156ad4c703b2c127b034abcR50

File: modules/components/src/DataTable/utils.js (hunk 1 above)
Comment (ciaranschutte) [reply modules/components/src/DataTable/utils.js]: yup tested :)

File: modules/components/src/DataTable/utils.js [hunk 3]
```diff
@@ -20,19 +20,33 @@ export function normalizeColumns({
...
+          : column.hasCustomType,
+        ...(!column.accessor && !column.id ? { id: column.field } : {}),
+        ...(customTypeConfigs[column.type] || {}),
+        ...customCol,
+      };
+    })
     .filter(x => x.show || x.canChangeShow);
-  return sortBy(customColumns, 'index').reduce(
+
+  // filter out override columns
```
Comment (hlminh2000) [inline modules/components/src/DataTable/utils.js]: this comment will really help in future! thanks :)

**Pull request merged**